prime-dotnet-webapi/Services/CommunitySiteService.cs

```diff
@@ -196,40 +196,41 @@ private void UpdateBusinessHours(Site current, CommunitySiteUpdateModel updated)
 
         private void UpdateRemoteUsers(Site current, CommunitySiteUpdateModel updated)
         {
-            // Wholesale replace the remote users
-            foreach (var remoteUser in current.RemoteUsers)
+            if (updated.RemoteUsers == null)
             {
-                foreach (var certification in remoteUser.RemoteUserCertifications)
-                {
-                    _context.Remove(certification);
-                }
-                _context.RemoteUsers.Remove(remoteUser);
+                return;
             }
 
-            if (updated?.RemoteUsers != null && updated?.RemoteUsers.Count() != 0)
+            foreach (var updateRemoteUser in updated.RemoteUsers)
             {
-                foreach (var remoteUser in updated.RemoteUsers)
-                {
-                    remoteUser.SiteId = current.Id;
-                    var remoteUserCertifications = new List<RemoteUserCertification>();
+                var existingRemoteUser = _context.RemoteUsers
```

Comment:
Feels like we could do this up front and not call out to the database on each iteration something like:

```c#
var existingRemoteUsers = await _context.RemoteUsers
    .Where(ru => ru.SiteId == current.Id)
    .Include(ru => ru.RemoteUserCertifications)
    .ToListAsync();

foreach (var updatedRemoteUser in updated.RemoteUsers)
{
    var existingRemoteUser = existingRemoteUsers.FirstOrDefault(ru => ru.Id == updatedRemoteUser.Id);
```